### PR TITLE
Test: Charge Deposition

### PR DESCRIPTION
**Pull request by ax3l**
Test the charge deposition logic & scaling of values.

- [x] fix charge (individual particle weighting) of MPI-parallel bunch init
- [x] add summing to MPI-enabled tests and generalize to work without duplicate cells as `sum()` does by default

Ran via
```console
cmake -S . -B build -DImpactX_PYTHON=ON
cmake --build build -j 2
cmake --build build -j 2 --target pip_install

# serial
python3 -m pytest -v -ssss tests/python/test_charge_deposition.py
# parallel
mpirun -n 4 python3 -m pytest -v -ssss tests/python/test_charge_deposition.py
```

Prepares #162
Depends on:
- https://github.com/AMReX-Codes/pyamrex/pull/50
- https://github.com/AMReX-Codes/pyamrex/pull/61
- https://github.com/AMReX-Codes/pyamrex/pull/63

### CODE REVIEW

File: tests/python/test_charge_deposition.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    beam_charge = dV*rs  # in C
+    # TODO: does not yet pass with MPI runs (too large)
+    assert math.isclose(beam_charge, 1.0e-9)
+
+    f = plt.figure()
+    ax = f.gca()
+    for mfi in rho:
+        bx = mfi.validbox()
+        #rbx = amrex.RealBox(bx, dr, gm.ProbLo())  # TODO
```
Comment (ax3l) [inline tests/python/test_charge_deposition.py]: Constructor to add to pyAMReX

Comment (ax3l) [reply tests/python/test_charge_deposition.py]: done in https://github.com/AMReX-Codes/pyamrex/pull/61

File: tests/python/test_charge_deposition.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    sim.set_slice_step_diagnostics(False)
+
+    sim.init_grids()
+    sim.init_beam_distribution_from_inputs()
+    sim.init_lattice_elements_from_inputs()
+
+    sim.evolve()
+
+    rho = sim.rho(lev=0)
+    # TODO: for cell-centered data, this double counts non-owned cells with MPI
```
Comment (ax3l) [inline tests/python/test_charge_deposition.py]:
```suggestion
    # TODO: for non-cell-centered data, this double counts non-owned cells with MPI
```
